### PR TITLE
FooBarQix Completed

### PR DESCRIPTION
**Pull request by ArmandsDulbinskis**

### CODE REVIEW

File: solutionAD/App/App.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+
+class App
+{
+
+    public $skaitlis;
```
Comment (OskarsOzols) [inline solutionAD/App/App.php]: Only English, please!

File: solutionAD/App/App.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    public function __construct($skaitlis)
+    {
+        $this->skaitlis = $skaitlis;
+    }
+
+
+    public function getResult()
+    {
+       // throw new Exception();
```
Comment (OskarsOzols) [inline solutionAD/App/App.php]: Dead comment

File: solutionAD/App/App.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        $nibba='';
+        $skaitlis3 = $this->skaitlis;
```
Comment (OskarsOzols) [inline solutionAD/App/App.php]: Variable naming.

File: solutionAD/tests/unit/SumCest.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    public function givenNegativeNumberReturnException(UnitTester $I)
+    {
+        $app = new App(7);
+        try {
+            $result = $app->skaitlis;
+            // $I->assert;
+
+            $I->assertGreaterThan(0, $result, 'Fail');
```
Comment (OskarsOzols) [inline solutionAD/tests/unit/SumCest.php]: Bad failed state description

File: solutionAD/tests/unit/SumCest.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            //$I->fail('Fail');
+        } catch (Throwable $e) {
+           echo $e->getMessage();
+        }
+    }
+
+    public function givenNonIntegerReturnException(UnitTester $I)
```
Comment (OskarsOzols) [inline solutionAD/tests/unit/SumCest.php]: Misleading function name or input argument

File: solutionAD/tests/unit/SumCest.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            $I->assertInternalType('int', $result, 'Fail:NonInteger');
+
+        } catch (Throwable $e) {
+            echo $e->getMessage();
+
+        }
+    }
+    public function returnTheGivenNumberAsAStringIfThereIsNoTransformationToDo(UnitTester $I)
+    {
+        $app = new App(15);
```
Comment (OskarsOzols) [inline solutionAD/tests/unit/SumCest.php]: But this arguments divides with 3 and 5.

File: solutionAD/tests/unit/SumCest.php [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+{
+    public function _before(UnitTester $I)
+    {
+    }
+
+    public function _after(UnitTester $I)
+    {
+    }
+
+//    public function sumTest(UnitTester $I)
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: commented-out code

File: solutionAD/tests/unit/SumCest.php [hunk 8]
```diff
@@ -0,0 +1,83 @@
+<?php
+include dirname(__DIR__, 2) . '/App/App.php';
+
+class sumCest
+{
+    public function _before(UnitTester $I)
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: empty functions

File: solutionAD/tests/unit/SumCest.php [hunk 9]
```diff
@@ -0,0 +1,83 @@
+<?php
+include dirname(__DIR__, 2) . '/App/App.php';
+
+class sumCest
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: Follow naming conventions: 
for start check https://www.php-fig.org/bylaws/psr-naming-conventions/

File: solutionAD/tests/unit/SumCest.php [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        $result = $app->getResult();
+        $I->assertEquals('Foo, Bar', $result);
+    }
+
+    public function givenNegativeNumberReturnException(UnitTester $I)
+    {
+        $app = new App(7);
+        try {
+            $result = $app->skaitlis;
+            // $I->assert;
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: commented out code

File: solutionAD/tests/unit/SumCest.php [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+            $I->assertGreaterThan(0, $result, 'Fail');
+            //$I->fail('Fail');
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: commented-out code

File: solutionAD/tests/unit/SumCest.php [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+            $result = $app->skaitlis;
+
+            $I->assertInternalType('int', $result, 'Fail:NonInteger');
+
+        } catch (Throwable $e) {
+            echo $e->getMessage();
+
+        }
+    }
+    public function returnTheGivenNumberAsAStringIfThereIsNoTransformationToDo(UnitTester $I)
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: vertical formatting - use phpstorm reformat feature

File: solutionAD/tests/unit/SumCest.php [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        $app = new App(15);
+        try {
+
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: vertical formatting - unnecessary blank line

File: solutionAD/tests/unit/SumCest.php [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+            $result = $app->getResult();
+            $dots = $app->skaitlis;
+            $I->assertNotEquals('',$result,$dots.'  There Is No Transformation To Do ');
+
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: vertical formatting - unnecessary blank line

File: solutionAD/tests/unit/SumCest.php [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+        } catch (Throwable $e) {
+            echo $e->getMessage();
+
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: vertical formatting - unnecessary blank line

File: solutionAD/tests/unit/SumCest.php [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+        }
+    }
+    public function returnTheGivenNumberAsAStringIfThereIsNoTransformationToDo(UnitTester $I)
+    {
+        $app = new App(15);
+        try {
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: obsolete try block

File: solutionAD/tests/unit/SumCest.php [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+            $result = $app->getResult();
+            $dots = $app->skaitlis;
```
Comment (enduro44) [inline solutionAD/tests/unit/SumCest.php]: Didn't understand idea

File: solutionAD/tests/unit/FooBarCest.php [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+        $app = new App(7);
+        try {
+            $result = $app->getResult();
+            $given = $app->number;
+            $I->assertNotEquals($given, $result, $result . ' There Is No Transformation To Do ');
+        } catch (Throwable $e) {
+            echo $e->getMessage();
+        }
+    }
+
```
Comment (enduro44) [inline solutionAD/tests/unit/FooBarCest.php]: blank line

File: solutionAD/tests/unit/FooBarCest.php [hunk 19]
```diff
@@ -0,0 +1,92 @@
+<?php
+include dirname(__DIR__, 2) . '/App/App.php';
+
+class FooBarCest
```
Comment (enduro44) [inline solutionAD/tests/unit/FooBarCest.php]: add qix

File: solutionAD/tests/unit/FooBarCest.php [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    public function givenNumberDividesWith5And7ReturnFooBar(UnitTester $I)
+    {
+        $app = new App(35);
+        $result = $app->getResult();
+        $I->assertEquals('Bar, Qix', $result);
+    }
+
+    public function givenNumberMustBePositive(UnitTester $I)
```
Comment (enduro44) [inline solutionAD/tests/unit/FooBarCest.php]: naming

File: solutionAD/tests/unit/FooBarCest.php [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+        $app = new App(7);
+        try {
+            $result = $app->number;
+            $I->assertInternalType('int', $result, 'Fail:Enter integer Number');
+        } catch (Throwable $e) {
+            echo $e->getMessage();
+        }
+    }
+
+    public function returnTheGivenNumberAsAStringIfThereIsNoTransformationToDo(UnitTester $I)
```
Comment (enduro44) [inline solutionAD/tests/unit/FooBarCest.php]: Refactor

File: .gitignore [hunk 22]
```diff
@@ -0,0 +1,30 @@
+
+# Default ignored files
+workspace.xml
+.gitkeep
+Acceptance.php
+acceptance.suite.yml
+AcceptanceTester.php
+AcceptanceTesterActions.php
+
+Functional.php
```
Comment (enduro44) [inline .gitignore]: Why these files are in gitignore?

File: .gitignore [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+FunctionalTesterActions.php
+misc.xml
+modules.xml
+
+solutionAD/tests/_support/Helper/Unit.php
+unit.suite.yml
+UnitTester.php
+UnitTesterActions.php
+vcs.xml
+vēlreiz.iml
```
Comment (enduro44) [inline .gitignore]: File should contain relevant ignores for all, not some local user specific file exclude

File: solutionAD/tests/unit/FooBarQixCest.php [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+    public function givenNumberIsNotIntegerThrowException(UnitTester $I)
+    {
+        $I->expectThrowable(Exception::class, function () {
+            (new App('7'))->getResult();
+        });
+    }
+
+    public function returnTheGivenNumberAsAStringIfThereIsNoTransformationToDo(UnitTester $I)
+    {
+        $I->expectThrowable(Exception::class, function () {
```
Comment (enduro44) [inline solutionAD/tests/unit/FooBarQixCest.php]: Wrong Bussiness logic - this case should return number as string no exception

File: solutionAD/tests/unit/FooBarQixCest.php [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+            (new App('7'))->getResult();
+        });
+    }
+
+    public function returnTheGivenNumberAsAStringIfThereIsNoTransformationToDo(UnitTester $I)
+    {
+        $app = new App(92);
+        $result = $app->getResult();
+        $I->assertIsString($result, 'There Is No Transformation To Do');
```
Comment (enduro44) [inline solutionAD/tests/unit/FooBarQixCest.php]: Better, but still wrong bussiness logic

File: solutionAD/App/App.php [hunk 26]
```diff
@@ -0,0 +1,114 @@
...
+    {
+        if (!is_int($this->number)) {
+            throw new Exception("FAIL:Enter integer Number");
+        }
+        if ($this->number < 0) {
+            throw new Exception("ERROR:Enter positive number");
+        }
+
+
+        if ($this->getMultiple() == '') {
```
Comment (enduro44) [inline solutionAD/App/App.php]: Move conditions to separate function

File: solutionAD/App/App.php [hunk 27]
```diff
@@ -0,0 +1,114 @@
...
+
+    public function __construct($number)
+    {
+        $this->number = $number;
+    }
+
+
+    private function getResultDividesBy3()
+    {
+        $number3 = $this->number;
```
Comment (enduro44) [inline solutionAD/App/App.php]: unnesassary variable

File: solutionAD/App/App.php [hunk 28]
```diff
@@ -0,0 +1,114 @@
...
+        $result = '';
+        $number3 %= 3;
+        if ($number3 == 0)
```
Comment (enduro44) [inline solutionAD/App/App.php]: if($this->number % 3 ===0)

File: solutionAD/App/App.php [hunk 29]
```diff
@@ -0,0 +1,114 @@
...
+        if ($number7 == 0)
+            $result .= 'Qix';
+        return $result;
+    }
+
+
+    public function getMultiple()
+    {
+        $result = '';
+        $result .= $this->getResultDividesBy3();
```
Comment (enduro44) [inline solutionAD/App/App.php]: change to array - map, filter, reduce, implode, explode

**Pull request closed without merging**